module/executiondatasync/optimistic_sync/persisters/stores/events.go

```diff
@@ -7,43 +7,37 @@ import (
 
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/storage"
-	"github.com/onflow/flow-go/storage/store/inmemory/unsynchronized"
 )
 
 var _ PersisterStore = (*EventsStore)(nil)
 
 // EventsStore handles persisting events
 type EventsStore struct {
-	inMemoryEvents  *unsynchronized.Events
+	data            []flow.Event
 	persistedEvents storage.Events
 	blockID         flow.Identifier
 }
 
 func NewEventsStore(
-	inMemoryEvents *unsynchronized.Events,
+	data []flow.Event,
 	persistedEvents storage.Events,
 	blockID flow.Identifier,
 ) *EventsStore {
 	return &EventsStore{
-		inMemoryEvents:  inMemoryEvents,
+		data:            data,
 		persistedEvents: persistedEvents,
 		blockID:         blockID,
 	}
 }
 
 // Persist adds events to the batch.
-// No errors are expected during normal operations
-func (e *EventsStore) Persist(lctx lockctx.Proof, batch storage.ReaderBatchWriter) error {
-	eventsList, err := e.inMemoryEvents.ByBlockID(e.blockID)
-	if err != nil {
-		return fmt.Errorf("could not get events: %w", err)
-	}
-
-	if len(eventsList) > 0 {
-		if err := e.persistedEvents.BatchStore(e.blockID, []flow.EventsList{eventsList}, batch); err != nil {
+//
+// No error returns are expected during normal operations
+func (e *EventsStore) Persist(_ lockctx.Proof, batch storage.ReaderBatchWriter) error {
+	if len(e.data) > 0 {
```

Comment:
we don't have to do this optimization. It's good to go through the storage, so that the storage knows the block has no event, and it updates the cache to speed up queries.